cli/internal/otel/receiver.go

```diff
@@ -145,6 +142,7 @@ func newDefaultMetricConsumer(metricsFile *os.File, quit chan any) ConsumeMetric
 		clientId := metric.Attributes["sync.client.id"].(string)
 
 		tableLock.Lock()
+		defer tableLock.Unlock()
```

Comment:
This ensures `renderTable` is handled only after we finished dealing with the latest batch